### ISSUE TITLE
Upgrade paypal integration (better_spree_paypal_express) - merchant-sdk-ruby or Checkout-Ruby-SDK

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/openfoodnetwork#5857]
### What we should change and why (this is tech debt)
We use better_spree_paypal_express v2.1.0 that uses paypal-sdk-merchant (which is [merchant-sdk-ruby](https://github.com/paypal/merchant-sdk-ruby)).
We are on paypal-sdk-merchant (= 1.106.1) from Oct 2013.
The last version of paypal-sdk-merchant v1.117.2 is from Aug 2015 and there is an indication of potential incompatibilities with rails 4.1

It looks like the sdk then evolved to latest rest api sdk: https://github.com/paypal/Checkout-Ruby-SDK

We have to estimate the task of upgrading to Checkout-Ruby-SDK....

### Context
This came up in https://github.com/openfoodfoundation/openfoodnetwork/issues/5855 as we thought maybe the old version we are using is causing this certificates problem in staging/sandbox. We are not sure these issues are related yet.

### Impact and timeline
Two reasons to upgrade now:
- this could be a rails 4.1 upgrade blocker...
- this can unblock resolve #5855 and make us be able to test paypal in staging again

**Comment by sauloperez** [comment]
I would add that the underlying PayPal SDK we're using bundling its own certificates (https://github.com/paypal/PayPal-Ruby-SDK/tree/v0.2.10/data) makes it riskier as we are locked in time. Any change on PayPal servers regarding SSL certificates and us not upgrading the SDK can be catastrophic. And I think that's what's happening with PayPal sandbox somehow.

**Comment by luisramos0** [comment]
yes :+1: I removed that cert from the gem on the staging server (maybe I didnt remove in the right place) but the error was still there.

**Comment by luisramos0** [comment]
The certificate issue is now solved. We need to test if paypal works in rails 4.1. at some point or not. Maybe this is not a problem for rails 4.1 :+1:

**Comment by rioug** [comment]
It looks like [activemerchant](https://github.com/activemerchant/active_merchant) supports Paypal,  maybe we should look into consolidating our use of activemerchant.

**Comment by rioug** [comment]
Paypal sdk merchant is deprecated : https://rubygems.org/gems/paypal-sdk-merchant/versions/1.117.4 
https://my.diffend.io/gems/paypal-sdk-merchant/1.117.2/1.117.3
